mongodb-connection-spike.md (#76)
percona-platform/portal-doc: docs/advisors/checks/mongodb-connection-spike.md

# MongoDB Connections sudden spike

## Description

> This check returns a warning if there is an increase in the number of connections that is higher than 50% of the most recent or normal number of connections.

Reviewer: ```suggestion
This check warns if the number of connections increases by 50% comparted to the most recent or normal number of connections.
```

Author: @catalinaadam 
1. There seems to be a typo. Instead of  "comparted" it should be "compared".
2. It statement should be more than 50% like this --- "increases by more than 50% compared to".

## Resolution

> There could be several reasons for seeing a sudden spike in MongoDB connections, including:

Reviewer: ```suggestion
A sudden spike in MongoDB connections can be caused by multiple factors, including:
```

> 1. **Increased User Traffic:** If there has been a sudden increase in the number of users accessing your MongoDB database, it could result in a spike in connections.

Reviewer: ```suggestion
1. **Increased User Traffic:** A sudden increase in the number of users accessing your MongoDB database could result in a connection spike.
```

> 2. **Database Maintenance:** If database maintenance activities such as backups, indexing, or data migrations are being performed, it could result in a spike in connections when a restart occurs or when application servers reconnect.

Reviewer: ```suggestion
2. **Database Maintenance:** Database maintenance activities such as backups, indexing, or data migrations could cause connections to spike when a restart occurs or when application servers reconnect.
```

> 3. **Poor Connection Pooling:** If your application is not using connection pooling properly, it could result in a spike in connections as new connections are created and old ones are not properly closed.

Reviewer: ```suggestion
3. **Poor Connection Pooling:** Applications not using connection pooling properly can cause connections to spike because this creates new connections without properly closing old ones. ```

> 4. **New batch jobs:** If there are new batch jobs added it can cause a spike in the number of connections.

Reviewer: ```suggestion
4. **New batch jobs:** New batch jobs added can cause a spike in the number of connections.
```

5. **Long-running Queries:** Long-running queries can also result in a spike in connections as they tie up resources, preventing new connections from being established. This could cause connections to queue.

> 6. **Poorly Configured Connection Limits:** If your MongoDB database is not configured with appropriate connection limits, it could result in a spike in the number of connections as new connections are created.

Reviewer: ```suggestion
6. **Poorly Configured Connection Limits:** MongoDB databases not configured with appropriate connection limits could cause connections to spike as new connections are created.
```

7. **Security Attacks:** Malicious actors can sometimes launch attacks to create a sudden surge of connections in an attempt to overwhelm and disrupt the database. An example of this would be Denial of Service (DOS) attacks. 

To diagnose the root cause of the sudden spike in connections, you should examine your logs and check related performance metrics to identify any unusual patterns or activity. This can help you determine if the issue is related to user traffic, application code, database configuration, or security issues.

> Some of the key Performance metrics that need to be examined are:

Reviewer: ```suggestion
Some of the key Performance metrics that you should examined are:
```
